packages/blade/src/components/Input/OTPInput/_decisions/_decisions.md

```diff
@@ -0,0 +1,93 @@
+# OTPInput Decisions <!-- omit in toc -->
+
+This doc talks about the API decisions for `OTPInput`. 
+## Index <!-- omit in toc -->
+
+- [OTPInput API](#otpinput-api)
+  - [Sample Usage](#sample-usage)
+- [Keyboard return key types for web and native](#keyboard-return-key-types-for-web-and-native)
+- [OTP SMS auto-read feature](#otp-sms-auto-read-feature)
+  - [Web](#web)
+  - [Native Android](#native-android)
+  - [Native iOS](#native-ios)
+
+## OTPInput API
+
+<img src="./otp.png" width="300" alt="otp input" />
+
+### Sample Usage
+
+```tsx
+<OTPInput
+  otpLength={4}
+  label="Enter One Time Code"
+  labelPosition="left"
+  helpText="Enter the OTP sent to your phone number"
+  errorText="Invalid OTP"
+  successText="Valid OTP"
+  validationState={inputValue.length < 4 ? 'error' : 'success'}
+/>
+```
+
+
+| Prop | Type | Required | Default | Description |
+|---|---|---|---|---|
+| otpLength | `4`, `6`  | No | `4` | Determines the length of the OTP input. |
+| label | `string` | Yes |  | Determines the label of the input field. Also used as `aria-label` |
+| labelPosition | `top`, `left` | No | `top` | Used to identify if the label of the input field will be placed on the top of the input field or left. `left` is only available for large screen devices. When position is `left` then the width of the label is fixed to `120px` and the content is aligned left. |
+| necessityIndicator | `optional`, `required` | No |  | Indicator to be shown next to the label for emphasizing the necessity of the field. if `optional` is provided then it'll show `${label} (optional)` and if `required` is provided then it'll show `${label} *` |
+| name | `string` | No |  | The name of the input, [useful in form submissions](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#name) |
+| value | `string` | No |  | Makes input field [controlled](https://reactjs.org/docs/forms.html#controlled-components) |
+| defaultValue | `string` | No |  | Used to set the default value of input field when it's [uncontrolled](https://reactjs.org/docs/uncontrolled-components.html#default-values) controlled |
+| validationState | `error` , `success`, `none` | No |  | Applies the style to the input field based on validation status and shows either `errorText` or `successText` respectively |
```

Comment:
Can this not be handled internally if it's based on otp length? Or are we letting consumer add custom validation also?

Reply:
It won't just be based on OTP length, right? A user could input an invalid OTP, and the consumer would also need to show that error. Its better to keep the API similar to rest of the TextInputs as well so that the familiarity and ease of use exists.